src/gcodePlanner.cpp

```diff
@@ -448,6 +448,80 @@ void GCodePlanner::addLinesByOptimizer(Polygons& polygons, GCodePathConfig* conf
+
+    if (last_wall_seam_vertex_idx >= 0)
+    {
+        // now we check that the vertex following the seam vertex is not to the right of the seam vertex in the last layer
+        // and if it is we move forward
+
+        // get the inward normal of the last layer seam vertex
+        Point last_wall_seam_vertex = last_wall[last_wall_seam_vertex_idx];
+        Point last_wall_seam_vertex_inward_normal = PolygonUtils::getVertexInwardNormal(last_wall, last_wall_seam_vertex_idx);
+
+        // create a vector from the normal so that we can then test the vertex following the candidate seam vertex to make sure it is on the correct side
+        Point last_wall_seam_vertex_vector = last_wall_seam_vertex + last_wall_seam_vertex_inward_normal;
+
+        // now test the vertex following the candidate seam vertex and if it lies to the left or on the vector, it's good to use
+        const int first_seam_vertex_idx = seam_vertex_idx;
+        float a = LinearAlg2D::getAngleLeft(last_wall_seam_vertex_vector, last_wall_seam_vertex, wall[(seam_vertex_idx + 1) % n_points]);
+        //std::cerr << layer_nr << ": angle = " << a * 180/M_PI << "\n";
+
+        while (a > M_PI)
+        {
+            // the vertex was on the right of the vector so move the seam vertex on
+            seam_vertex_idx = (seam_vertex_idx + 1) % n_points;
+            a = LinearAlg2D::getAngleLeft(last_wall_seam_vertex_vector, last_wall_seam_vertex, wall[(seam_vertex_idx + 1) % n_points]);
+            //std::cerr << "new angle = " << a * 180/M_PI << "\n";
+
+            if (seam_vertex_idx == first_seam_vertex_idx)
+            {
+                // this shouldn't happen!
+                break;
+            }
+        }
+    }
+
+    Polygons last_wall_polygons;
+    last_wall_polygons.add(last_wall);
+    const int max_dist = MM2INT(config->getLineWidth() * 5);
+    // extrude to the points following the seam vertex
+    // the last point is the seam vertex as the polygon is a loop
+    for (int i = 1; i <= n_points; ++i)
+    {
+        // p is a point from the current wall polygon
+        const Point& p = wall[(seam_vertex_idx + i) % n_points];
+        // last_p is p shifted onto last_wall
+        Point last_p(p);
+        if (last_wall.inside(last_p, false))
+        {
+            PolygonUtils::moveOutside(last_wall_polygons, last_p, 0, max_dist);
+        }
+        else if(!last_wall.inside(last_p, true))
+        {
+            PolygonUtils::moveInside(last_wall_polygons, last_p, 0, max_dist);
+        }
+        // now interpolate between last_p and p depending on how far we have progressed along wall
+#if 0
+        if(i == n_points)
+            addTravel_simple(p);
+        else
+#endif
+        addExtrusionMove(Point(last_p + (p - last_p) * static_cast<double>(i) / n_points), config, SpaceFillType::Polygons, 1.0, true);
```

Comment:
Please introduce a variable `coord_t polygon_length = wall.length()` and a variable `polygon_length_traversed`which you update on every iteration of this for loop, so that instead of doing `* i / n_points` you do `* polygon_length_traversed / polygon_length`.